### PR TITLE
Clarify truncation with addresses

### PR DESCRIPTION
**Pull request by ChrisChinchilla**
As mentioned here - 
https://github.com/ethereum/solidity/issues/4341#issuecomment-402542639

Hopefully I got this right.

### CODE REVIEW

File: docs/types.rst [hunk 1]
```diff
@@ -104,6 +104,12 @@ Operators:
 
 * ``<=``, ``<``, ``==``, ``!=``, ``>=`` and ``>``
 
+.. warning::
+  If you convert a type that uses a larger byte size to an ``address``, for example ``byte32``, then the ``address`` is truncated. To reduce conversion ambiguity version 0.4.24 and higher of the compiler force you make the truncation explicit in the conversion. Take for example the address ``0x111122223333444455556666777788889999AAAABBBBCCCCDDDDEEEEFFFFCCCC``.
```
Comment (chriseth) [inline docs/types.rst]: Is `0x111122223333444455556666777788889999AAAABBBBCCCCDDDDEEEEFFFFCCCC` an address? It is only a `bytes32` literal, isn't it?

File: docs/types.rst [hunk 2]
```diff
@@ -104,6 +104,12 @@ Operators:
...
+
+  You can use ``address(uint160(bytes20(b)))``, which results in 0x111122223333444455556666777788889999aAaa,
```
Comment (chriseth) [inline docs/types.rst]: Please also put these literals in double backticks.

File: docs/types.rst (hunk 1 above)
Comment (ChrisChinchilla) [reply docs/types.rst]: @chriseth I'm not sure I have the knowledge yet to say for sure. Maybe @ekpyron could comment here as this was paraphrased from a comment they made.

File: docs/types.rst (hunk 2 above)
Comment (ChrisChinchilla) [reply docs/types.rst]: 👍 Done.

File: docs/types.rst (hunk 1 above)
Comment (axic) [inline docs/types.rst]: `byte32` -> `bytes32`

Comment (ChrisChinchilla) [reply docs/types.rst]: @axic Done

**Comment by chriseth** [comment]
@ChrisChinchilla sorry, we did not really explain what this is all about. Please squash and merge if you are fine with my changes.

**Comment by ChrisChinchilla** [comment]
@chriseth Done!

**Review by axic — APPROVED** [review]

**Pull request merged**